dask/array/tests/test_ufunc.py

```diff
@@ -474,3 +474,20 @@ def test_divmod():
     expected = divmod(arr1, arr2)
     assert_eq(result[0], expected[0])
     assert_eq(result[1], expected[1])
+
+
+@pytest.mark.parametrize("dt", ["float64", "float32", "int32", "int64"])
+def test_dtype_kwarg(dt):
+    x1 = da.asarray([1, 2, 3])
+    y1 = da.asarray([4, 5, 6])
+    z1 = np.add(x1, y1, dtype=np.dtype(dt))
+    assert z1.dtype == np.dtype(dt)
+    assert z1.compute().dtype == z1.dtype
+
+    x2 = da.asarray([1, 2, 3])
+    y2 = da.asarray([4, 5, 6])
+    z2 = da.add(x2, y2, dtype=np.dtype(dt))
+    assert z2.dtype == np.dtype(dt)
+    assert z2.compute().dtype == z2.dtype
+
+    assert_eq(z1, z2)
```

Comment:
This test can be simplified because `assert_eq` already covers the dtype check. To convince yourself you can pull this change, comment out the `enforce_dtype = True` line from `core.py` and then run the test and watch it fail. 

This is my suggestion:
```suggestion
    arr1 = np.array([1, 2, 3])
    arr2 = np.array([4, 5, 6])

    darr1 = da.from_array(arr1)
    darr2 = da.from_array(arr2)

    expected = np.add(arr1, arr2, dtype=dt)
    result = np.add(darr1, darr2, dtype=dt)
    assert_eq(expected, result)

    result = da.add(darr1, darr2, dtype=dt)
    assert_eq(expected, result)
```